File: perfectSubArray.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)

var reader *bufio.Reader

func solve(caseNo int) {
	n, err := strconv.ParseInt(readLine(reader), 10, 32)
	checkError(err)
	nTemps := strings.Split(readLine(reader), " ")
	var sum int
	sumFreqs := make(map[int]int)
	sumFreqs[0] = 1
	var count int64
	for itr := 0; itr < int(n); itr++ {
		curr, err := strconv.ParseInt(nTemps[itr], 10, 32)
		checkError(err)
		sum += int(curr)
		for i := 0; i*i < 1e7; i++ {
			count += int64(sumFreqs[sum-i*i])
		}
		val := sumFreqs[sum]
		sumFreqs[sum] = val + 1
	}
	fmt.Print("Case #" + strconv.Itoa(caseNo) + ": " + strconv.FormatInt(count, 10) + "\n")
}

func main() {
	reader = bufio.NewReaderSize(os.Stdin, 1024*1024)
	t, err := strconv.ParseInt(readLine(reader), 10, 32)
	checkError(err)
	for i := 1; i <= int(t); i++ {
		solve(i)
	}
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}

File: perfectSubArray.java
import java.util.*;
import java.io.*;
import java.math.*;
import static java.lang.System.*;


class Temp implements Runnable{
    static StringBuilder ans;
    static String[] nTemps;
    static HashMap<Integer, Integer> sumFreqs;
    public static void solve(int caseNo) throws IOException{
        nTemps = br.readLine().split(" ");
        int n = Integer.parseInt(nTemps[0]);
        nTemps = br.readLine().split(" ");
        int sum = 0, mSum = 0;
        sumFreqs.put(0, 1);
        long count = 0;
        for(int itr = 0; itr < n; itr++){
            int curr = Integer.parseInt(nTemps[itr]);
            sum+=curr;
            mSum = Math.min(mSum, sum);
            for(int i = 0; sum-i*i >= mSum; i++){
                count+=sumFreqs.getOrDefault(sum-(i*i), 0);
            }
            sumFreqs.put(sum, sumFreqs.getOrDefault(sum, 0)+1);
        }
        ans.append("Case #"+caseNo+": "+String.valueOf(count)+"\n");
    }
    final static BufferedReader br = new BufferedReader(new InputStreamReader(in));
    @Override
    public void run(){
        try{
            ans = new StringBuilder();
            int t = Integer.parseInt(br.readLine());
            sumFreqs = new HashMap<>();
            for(int itr = 1; itr <= t; itr++){
                sumFreqs.clear();
                solve(itr);    
            }
            out.print(ans);
            br.close();
        }catch(Exception e){
            Thread t = Thread.currentThread();
            t.getUncaughtExceptionHandler().uncaughtException(t, e);
        }
    }
}

class Solution {
    public static void main(String[] args) throws Exception
    {
        Thread t = new Thread(null, new Temp(), "Temp", 1 << 30);
        t.start();
        t.join();
    }
}